### PR TITLE
PE-57 Open API descriptor for CGN operator search API

### PR DESCRIPTION
**Pull request by ltronky**
#### List of Changes
Update OpenApi descriptor. Define endpoint to:
- retrieve the list of online merchants
- retrieve the list of offline merchants
- retrieve the details of a merchant together with its published discounts

#### Motivation and Context
Definition of API contracts

#### How Has This Been Tested?
N/A

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Review by francescopellegrini — DISMISSED** [review]

File: openapi/index.yaml [hunk 1]
```diff
@@ -1,49 +1,265 @@
...
+basePath: "/api/v1"
 schemes:
   - https
 paths:
-  "/echo/{input}":
+  /online-merchants:
     parameters:
-      - name: input
+      - "$ref": "#/parameters/Name"
+      - "$ref": "#/parameters/ProductCategories"
```
Comment (AleDore) [inline openapi/index.yaml:20]: We should move `parameters` into each operation and repeat them, due to our codegen compatibility

File: openapi/index.yaml [hunk 2]
```diff
@@ -1,49 +1,265 @@
...
+        type: string
+      description:
+        type: string
+      startDate:
+        type: string
+        format: date
+      endDate:
+        type: string
+        format: date
+      productCategories:
```
Comment (AleDore) [inline openapi/index.yaml:155]: I'm not sure but I'd rather check if `productCategories` items are enums :)

File: openapi/index.yaml [hunk 3]
```diff
@@ -1,49 +1,265 @@
...
     properties:
       id:
         type: string
-      output:
+      name:
+        type: string
+      productCategories:
+        type: array
+        items:
+          type: string
```
Comment (AleDore) [inline openapi/index.yaml:224]: ditto :)

File: openapi/index.yaml (hunk 1 above)
Comment (ltronky) [reply openapi/index.yaml]: I've moved the parameter definition

File: openapi/index.yaml (hunk 2 above)
Comment (ltronky) [reply openapi/index.yaml:175]: now the product categories are defined as enum aligned with specs

**Review by AleDore — APPROVED** [review]
LGTM

**Pull request merged**